CoreStoreTests/MigrationTests.swift

```diff
@@ -0,0 +1,235 @@
+//
+//  MigrationChainTests.swift
+//  CoreStore
+//
+//  Copyright © 2018 John Rommel Estropia
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in all
+//  copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+//  SOFTWARE.
+//
+
+import XCTest
+
+@testable
+import CoreStore
+
+
+// MARK: - MigrationTests
+
+final class MigrationTests: BaseTestCase {
+    
+    func test_ThatEntityDescriptionExtension_CanMapAttributes() {
+        
+        // Should match attributes by renaming identifier.
+        do {
+            let src = NSEntityDescription([NSAttributeDescription("foo")])
+            let dst = NSEntityDescription([NSAttributeDescription("bar", renamingIdentifier: "foo")])
+            
+            var map: [NSAttributeDescription: NSAttributeDescription] = [:]
+            XCTAssertNoThrow(map = try dst.mapAttributes(in: src))
+            XCTAssertEqual(map.count, 1)
+            XCTAssertEqual(map.keys.first?.renamingIdentifier, "foo")
+            XCTAssertEqual(map.values.first?.name, "foo")
+        }
+        
+        // Should match attributes by name when matching by renaming identifier fails.
+        do {
+            let src = NSEntityDescription([NSAttributeDescription("bar")])
+            let dst = NSEntityDescription([NSAttributeDescription("bar", renamingIdentifier: "foo")])
+            
+            var map: [NSAttributeDescription: NSAttributeDescription] = [:]
+            XCTAssertNoThrow(map = try dst.mapAttributes(in: src))
+            XCTAssertEqual(map.count, 1)
```

Comment:
In my understanding, in this case `mapAttributes()` should throw. The new `dst.bar` should be completely unrelated to `src.bar` (`dst.bar` should be linked to a `src.foo`)